src/cql/r4/Factors_to_Consider_in_Managing_Chronic_Pain_FHIRv400.cql

```diff
@@ -946,26 +953,184 @@ define ReportStoolSoftenersAndLaxatives:
+      rxNormCode: FHIRHelpers.ToCode((O.medication as FHIR.CodeableConcept).coding[0]),
+      medicationName: OMTKLogic.GetMedicationName(rxNormCode),
+      // NOTE: Assuming a single dosage instruction element
+      dosageInstruction: O.dosageInstruction[0],
+      // NOTE: Assuming a single dose and rate element
+      doseAndRate: dosageInstruction.doseAndRate[0],
+      repeat: dosageInstruction.timing.repeat,
+      frequency: Coalesce(repeat.frequencyMax.value, repeat.frequency.value),
+      period: System.Quantity { value: repeat.period.value, unit: repeat.periodUnit.value },
+      doseDescription:
+        Coalesce(
+          // There should be a conversion from FHIR.SimpleQuantity to System.Quantity
+          if doseAndRate.dose is FHIR.Range
+            then ToString((doseAndRate.dose as FHIR.Range).low)
+                + '-' + ToString((doseAndRate.dose as FHIR.Range).high)
+                + (doseAndRate.dose as FHIR.Range).high.unit.value
+            else ToString(FHIRHelpers.ToQuantity(doseAndRate.dose as FHIR.SimpleQuantity)),
+            ''
+        ),
+      frequencyDescription:
+        ToString(dosageInstruction.timing.repeat.frequency.value) +
+          Coalesce(
+            '-' + ToString(dosageInstruction.timing.repeat.frequencyMax.value),
+            ''
+          )
+    return {
+      rxNormCode: rxNormCode,
+      isDraft: O.status.value = 'draft',
+      // NOTE: Assuming asNeeded is expressed as a boolean
+      isPRN: dosageInstruction.asNeeded,
+      prescription:
+        if dosageInstruction.text is not null then
+          medicationName + ' ' + dosageInstruction.text.value
+        else
+          // TODO: Shouldn't need the .value here on asNeededBoolean
+          medicationName + ' ' + doseDescription + ' q' + frequencyDescription + (if dosageInstruction.asNeeded then ' PRN' else ''),
+      dose: if doseAndRate.dose is FHIR.Range
+            then FHIRHelpers.ToQuantity((doseAndRate.dose as FHIR.Range).high)
+            else FHIRHelpers.ToQuantity(doseAndRate.dose as FHIR.SimpleQuantity),
+      dosesPerDay: Coalesce(OMTKLogic.ToDaily(frequency, period), 1.0)
     }
 
-// Urine drug screen - dates and results (all screens w/ lookback of 1 year)
 
+define function MME(prescriptions List<MedicationRequest>):
+  (Prescriptions(prescriptions)) P
+    let mmePerIngredient: OMTKLogic.CalculateMMEs({ { rxNormCode: P.rxNormCode, doseQuantity: P.dose, dosesPerDay: P.dosesPerDay } })
+    return {
+      rxNormCode: P.rxNormCode,
+      isDraft: P.isDraft,
+      isPRN: P.isPRN,
+      prescription: P.prescription,
+      dailyDose: Combine(mmePerIngredient X return X.dailyDoseDescription, '\r\n'),
+      mme: Sum(mmePerIngredient X return X.mme)
+    }
+
+
+define function TotalMME(prescriptions List<MedicationRequest>):
+  System.Quantity {
+    value: Sum((MME(prescriptions)) M return M.mme.value),
+    unit: 'mg/d'
+  }
+
+
+define ReportMME:
+  return TotalMME(Prescriptions(Orders))
```

Comment:
I'm seeing the below error, from this line:
```
================================================================================
TRANSLATE /opt/cql/r4/Factors_to_Consider_in_Managing_Chronic_Pain_FHIRv400.cql
Translation failed due to errors:
Error:[1129:2, 1129:2] Syntax error at return
```

Reply:
This is a `define`, so I don't think it `return`s like `function`s do.

`Orders` wasn't defined, so I tried using `OpioidMedicationRequests`

```suggestion
  TotalMME(OpioidMedicationRequests)
```

I also had to remove the function call to `Prescriptions()` to get the CQL to translate. Otherwise, I got the below error:

```
================================================================================
TRANSLATE /opt/cql/r4/Factors_to_Consider_in_Managing_Chronic_Pain_FHIRv400.cql
Translation failed due to errors:
Error:[1130:3, 1130:51] Could not resolve call to operator TotalMME with signature (list<tuple{rxNormCode:System.Code,isDraft:System.Boolean,isPRN:choice<FHIR.boolean,FHIR.CodeableConcept>,prescription:System.String,dose:System.Quantity,dosesPerDay:System.Decimal}>).
Error:[1357:20, 1357:28] Could not validate reference to expression ReportMME because its definition contains errors.
```